projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,667 @@
+     *
+     * @remarks
+     * Emitted args are of type `IBaseEventArgs`
+     *
+     * @example
+     * ```html
+     * <igx-date-range (onClosed)="handleClosed($event)"></igx-date-range>
+     * ```
+     */
+    @Output()
+    public onClosed = new EventEmitter<IBaseEventArgs>();
+
+    /** @hidden */
+    @HostBinding('class.igx-date-range')
+    public cssClass = 'igx-date-range';
+
+    /** @hidden */
+    @ViewChild(IgxDateSingleComponent)
+    public single: IgxDateSingleComponent;
+
+    /** @hidden */
+    @ViewChild(IgxCalendarComponent)
+    public calendar: IgxCalendarComponent;
+
+    /** @hidden */
+    @ViewChild(IgxToggleDirective)
+    public toggleDirective: IgxToggleDirective;
+
+    /** @hidden */
+    @ContentChildren(IgxPickerToggleComponent, { descendants: true })
+    public toggleComponents: QueryList<IgxPickerToggleComponent>;
+
+    /** @hidden */
+    @ContentChildren(IgxInputGroupBase)
+    public projectedInputs: QueryList<IgxInputGroupBase>;
+
+    @ContentChild(IgxLabelDirective)
+    public label: IgxLabelDirective;
+
+    /** @hidden */
+    public prepTo = CurrentResourceStrings.RangeDatePickerResStrings.igx_range_date_picker_to;
+
+    /** @hidden @internal */
+    public get appliedFormat() {
+        // TODO: apply formatter if present
+        if (!this.hasProjectedInputs) {
+            // TODO: use displayFormat - see how shortDate, longDate can be defined
+            return this.inputFormat
+                ? `${this.inputFormat} - ${this.inputFormat}`
+                : `${DEFAULT_INPUT_FORMAT} - ${DEFAULT_INPUT_FORMAT}`;
+        } else {
+            return this.inputFormat ? this.inputFormat : DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public get hasProjectedInputs(): boolean {
+        return this.projectedInputs.some(i => i instanceof IgxDateStartComponent || i instanceof IgxDateEndComponent);
+    }
+
+    private _value: DateRange;
+    private $destroy = new Subject();
+    private $toggleClickNotifier = new Subject();
+    private _onChangeCallback: (_: any) => void;
+    private _positionSettings: PositionSettings;
+    private _dialogOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: true,
+        positionStrategy: new GlobalPositionStrategy()
+    };
+    private _dropDownOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: false
+    };
+
+    constructor(public element: ElementRef) {
+        this._onChangeCallback = (_: any) => { };
+    }
+
+    /**
+     * Opens the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range #dateRange mode="dialog"></igx-date-range>
+     *
+     * <button (click)="dateRange.open()">Open Dialog</button
+     * ```
+     */
+    public open(): void {
+        const range: Date[] = [];
+        if (this.value) {
+            if (this.value.start) {
```

Comment:
IIRC when we have no `start` we reset `value`.
If this is true I would put the check for `end` nested in the check for `start` and will clear the `value` otherwise